tests/main_test.go

```diff
@@ -0,0 +1,94 @@
+package tests
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/openfaas/faas-cli/commands"
+
+	sdkConfig "github.com/openfaas/faas-cli/config"
+
+	sdk "github.com/openfaas/faas-cli/proxy"
+)
+
+var (
+	config           = Config{}
+	defaultNamespace = ""
+	swarm            = flag.Bool("swarm", false, "helper flag to run only swarm-compatible tests only")
+	token            = flag.String("token", "", "authentication Bearer token override, enables auth automatically")
+)
+
+func init() {
+
+	flag.StringVar(&config.Gateway, "gateway", "", "set the gateway URL, if empty use the gateway_url env variable")
+
+	flag.BoolVar(
+		&config.AuthEnabled,
+		"enableAuth",
+		false,
+		fmt.Sprintf("enable/disable authentication. The auth will be parsed from the default config in %s", filepath.Join(sdkConfig.DefaultDir, sdkConfig.DefaultFile)),
+	)
+	flag.BoolVar(&config.SecretUpdate, "secretUpdate", true, "enable/disable secret update tests")
+	flag.BoolVar(&config.ScaleToZero, "scaleToZero", true, "enable/disable scale from zero tests")
+}
+
+func TestMain(m *testing.M) {
+	// flag parsing here
+	flag.Parse()
+
+	if config.Gateway == "" {
+		uri, err := url.Parse(os.Getenv("gateway_url"))
```

Comment:
Please could you use `os.LookupEnv()` here instead? I'll merge, but can accept a follow-up patch.